crypto/ec/ecx_backend.c

```diff
@@ -18,6 +20,31 @@
  * implementations alike.
  */
 
+int ecx_public_from_private(ECX_KEY *key)
+{
+    switch (key->type) {
+    case ECX_KEY_TYPE_X25519:
+        X25519_public_from_private(key->pubkey, key->privkey);
+        break;
+    case ECX_KEY_TYPE_ED25519:
+        if (!ED25519_public_from_private(key->libctx, key->pubkey, key->privkey)) {
```

Comment:
Should then functions be passing a property_query string also?

Reply:
This is definitely a non-trivial task. The problem is that this code is called provider side. If we're going to use the property query here then we'd have to pass it through as an OSSL_PARAM, and presumably have an `ecx_set_params` implementation in order to be able to associate that query with the key in some way. I suspect this problem is bigger than just ECX (we're quite inconsistent about where we pass both a libctx and propq and where we just pass a libctx) - so I suspect we're going to need to address that as a separate issue. Solving this in this case is straying quite far from my original scope for this PR - so I'd rather not try and deal with this here.

Reply:
We pass property queries without the libctx in places.  KDF definitely but I think MAC too.